### ISSUE TITLE
quality issue because of messy commits

### ISSUE THREAD

**Issue by tomhawks** [GenisysPro/GenisysPro#1106]
Hi guys,

I have recently noticed GenisysPro is losing quality because many people are ruining it. However, @CortexPE is making are software very messy & unusable with his hacks

For example:
- Mirroring the packets which was a broken hack
- The infamous getName issue

Please stop merging all these bad hacks from cortexpe ruining the genisyspro software. Thanks.

Cortexpe: Please stop making hacks

**Comment by SuperXingKong** [comment]
Yeah we really need to improve the quality of pulls.
Partly because I'm lazy, on the other hand, I do have very little time because my schoolwork.
Firstly, I should blame myself.

**Comment by FaigerSYS** [comment]
Also this (for example)
#1100

**Comment by InspectorGadget** [comment]
I agree with FaigerSYS, I don't get the logic behind copying. It's hilarious on how people copy blindly from PMMP and then add a getName() and then claim that you've done stuffs but no in reality.

**Comment by tomhawks** [comment]
@SuperXingKong they work very hard to keep genisyspro stable & usable for all, thanks for making genisyspro to allow this great software to continue 👍 

I believe we should start a major cleanup task before updating to 1.2, and stop copy pasting/merging hacks.

**Comment by tomhawks** [comment]
@FaigerSYS @InspectorGadget agreed. everyone is copy pasting bad code and ruining our project :(

Vote 👍or👎 if you think genisyspro is messy

**Comment by InspectorGadget** [comment]
Yes, a cleanup is needed. I can see obfuscated code. I'll help throughout the process wherever needed. At the same time, we need other members' commitment too.

**Comment by tomhawks** [comment]
we should start by kicking off members who only contribute hacks, add more strict types, implement the new pmmp network handlers .

maybe stricter pull request checks

this is the beginging of a whole new advancement. 👍

**Comment by InspectorGadget** [comment]
Yes, strict type implementation is a must. We should also add the network handlers. This applies for Pull Requests too.

**Comment by MasterOfTheRealm** [comment]
Fyi CortexPE is only trying to make GenisysPro better. The "infamous" getName() issue was resolved before it became a big deal to the software, but a lifeless entity called "Betty" has been spamming issue trackers of all softwares with that issue making it so called "infamous".
As for mirroring packets... It would have not affected anyone using the network/mcpe/protocol path.

**Comment by tomhawks** [comment]
@masteroftherealm if you idea to make genisyspro better is by adding disgusting hacks, crap behavior will happen such as crashes.

Now: mirroring packets also has caused a start up crash for me until SuperXingKong fixed it. I’m not saying cortex pe is a bad dev or is trying to ruin genisys. I’m saying his code is crap and the consequences are fucking spam on every pmmp repo and he shouldn’t be using hacks

**Comment by MasterOfTheRealm** [comment]
Again it's Betty who spams the pmmp repo... Blame Betty not Cortex. And even if he did make bad commits once in a while he has helped more that most people in here.

**Comment by xISRAPILx** [comment]
O'kay...
